data_model/src/lib.rs

```diff
@@ -745,6 +745,16 @@ pub trait Registered: Identifiable {
     type With: Into<RegistrableBox>;
 }
 
+/// Trait for objects used to create target objects
+#[cfg(feature = "mutable_api")]
+pub trait Builder {
+    /// Constructed type
+    type Target;
+
+    /// Construct [`Self::Target`]
+    fn build(self) -> Self::Target;
+}
```

Comment:
What do you think about this:

```suggestion
pub trait Registrable {
    /// Constructed type
    type Target: Identifiable;

    /// Identification of [`Self::Target`]
    fn target_id(&self) -> <Self::Target as Identifiable>::Id;

    /// Construct [`Self::Target`]
    fn build(self) -> Self::Target;
}
```

Reply:
it does make sense to combine them into one trait. `Builder` and `Register` always go together, at least at the moment. I don't think we need `fn target_id` though

Reply:
As i find out there some cases where type implement `Registered<With = Self>` (Trigger, Peer, Account).
In this case having single trait `Builder + Registered` seems inconvenient.
However @s8sato suggestion is still applicable.